Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +447,134 @@ def restore_snapshot_online(
+        expirationHours = convert_to_hours(expiration_time, expiration_unit)
+        retentionHours = convert_to_hours(retention_time, retention_unit)
+        if not client_obj.scheduleExists(schedule_name):
+           cmd = ["createsv"]
+           if read_only:
+              cmd.append("-ro")
+              cmd.append("-exp")
+              cmd.append(str(expirationHours))
+              cmd.append("-retain")
+              cmd.append(str(retentionHours))
+              snap_string = ".@y@@m@@d@@H@@M@@S@"
+              cmd.append("snap-"+base_volume_name+snap_string)
+              cmd.append(base_volume_name)
+              if task_freq_custom:
+                 freq = task_freq_custom
+
+              if task_freq:
+                  freq="@"+task_freq
+              cmd = ' '.join(cmd)
+              client_obj.createSchedule(
+                schedule_name, cmd, freq)
+        else:
+            return (True, False, "Schedule not Exist", {})
+    except Exception as e:
+        return (False, "False", "Schedule creation failed | %s" % (e), {})
+    finally:
+        client_obj.logout()
+    return (
+        True,
+        True,
+        "Created Schedule %s successfully." %
+        schedule_name,
+        {})
+                
+def delete_schedule(
+        client_obj,
+        storage_system_ip,
+        storage_system_username,
+        storage_system_password,
+        schedule_name):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "schedule delete failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "schedule delete failed. Schedule name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "schedule create failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        if client_obj.scheduleExists(schedule_name):
+           client_obj.deleteSchedule(schedule_name)
+        else:
+           return (True, False, "Schedule does not exist", {})
+    except Exception as e:
+        return (False, False, "Schedule delete failed | %s" % (e), {})
+    finally:
+        client_obj.logout()
+    return (
+        True,
+        True,
+        "Deleted Schedule %s successfully." %
+        schedule_name,
+        {})             
 
 def main():
 
     fields = {
         "state": {
             "required": True,
-            "choices": ['present', 'absent', 'modify', 'restore_offline',
+            "choices": ['present', 'absent', 'schedule_create', 'schedule_delete', 'modify', 'restore_offline',
```

Comment:
Please name these create_schedule and delete_schedule

Reply:
should i renamed it create and delete?

Reply:
No, create_schedule and delete_schedule